internal/sbi/producer/nf_discovery.go

```diff
@@ -38,10 +38,33 @@ func HandleNFDiscoveryRequest(request *httpwrapper.Request) *httpwrapper.Respons
 	return httpwrapper.NewResponse(http.StatusForbidden, nil, problemDetails)
 }
 
+func ValidateQueryParameters(queryParameters url.Values) bool {
+	NFs := [...]string{"BSF", "SMF", "UPF", "PCF", "AMF", "CHF", "AUSF", "UDM", "UDR", "NSSF"}
+	tar := queryParameters["target-nf-type"][0]
+	req := queryParameters["requester-nf-type"][0]
+	result0 := false
+	result1 := false
+
+	for _, nf := range NFs {
+		if tar == nf {
+			result0 = true
+		}
+		if req == nf {
+			result1 = true
+		}
+	}
+	supi_check := true
+	if queryParameters["supi"] != nil {
+		supi_check = (queryParameters["supi"][0][0:5] == "imsi-")
```

Comment:
```
if queryParameters["supi"][0][:5] != "imsi-" {
    return false
}
```

Reply:
line 57 is necessary, since supi is not present everytime. otherwise there will be out of index error